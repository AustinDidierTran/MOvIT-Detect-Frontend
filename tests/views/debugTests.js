/**
 * @author Gabriel Boucher
 * @author Anne-Marie Desloges
 * @author Austin-Didier Tran
 * @author Benjamin Roy
 */

import Enzyme, { shallow } from 'enzyme';

import Adapter from 'enzyme-adapter-react-16';
import Debug from '../../src/views/debug';
import PropTypes from 'prop-types';
import React from 'react';
import configureMockStore from 'redux-mock-store';
import toJson from 'enzyme-to-json';

Enzyme.configure({ adapter: new Adapter() });

describe('Debug Tests', () => {
  const initialState = { applicationReducer: { language: 'en' } };
  const mockStore = configureMockStore();
  const store = mockStore(initialState);
  const props = {};

  it('should have proptypes', () => {
    const actualValue = Debug.WrappedComponent.propTypes;

    const expectedValue = {
      language: PropTypes.string.isRequired,
    };

    expect(JSON.stringify(actualValue)).toEqual(JSON.stringify(expectedValue));
  });

  it('should match the snapshot', () => {
    const wrapper = shallow(<Debug store={store} {...props} />).dive();

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
