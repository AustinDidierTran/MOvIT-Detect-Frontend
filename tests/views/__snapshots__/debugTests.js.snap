// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Debug Tests should match the snapshot 1`] = `
<div
  className="container"
>
  <div
    className="row mt-4"
  >
    <div
      className="col-12"
    >
      <h2
        className="header text-center"
      >
        debug.state.en
      </h2>
      <Connect(Notification) />
      <Connect(NotificationSettings) />
      <Connect(UpdatesManager) />
    </div>
  </div>
</div>
`;
